pypeit/reduce.py

```diff
@@ -645,60 +660,115 @@ def load_skyregions(self, skymask_init):
                 usersky = True
         return skymask_init, usersky
 
-    def spec_flexure_correct(self, sobjs, basename):
+    def spec_flexure_correct(self, basename, mode="object", specObjs=None, update_waveimg=False):
         """ Correct for spectral flexure
 
         Spectra are modified in place (wavelengths are shifted)
 
-        Wrapper to flexure.flexure_obj()
-
         Args:
-            sobjs (:class:`pypeit.specobjs.SpecObjs`):
             basename (str):
-
+            mode (str):
+                "object" - Use specObjs to determine flexure correction
+                "slit" - Use waveimg and global_sky to determine flexure correction at the centre of the slit
+            specObjs (:class:`pypeit.specobjs.SpecObjs`, None):
+                Spectrally extracted objects
         """
         if self.par['flexure']['spec_method'] != 'skip':
-            # Measure
-            flex_list = flexure.spec_flexure_obj(sobjs, self.slits.slitord_id, self.reduce_bpm,
-                                                 self.par['flexure']['spec_method'],
-                                                 self.par['flexure']['spectrum'],
-                                                 mxshft=self.par['flexure']['spec_maxshift'])
-            # QA
-            flexure.spec_flexure_qa(sobjs, self.slits.slitord_id, self.reduce_bpm, basename, self.det, flex_list,
-                                    out_dir=os.path.join(self.par['rdx']['redux_path'], 'QA'))
+            # Initialise
+            slitSpecs = None
+            gd_slits = np.logical_not(self.reduce_bpm)
+            wvimg = self.waveimg.copy() if update_waveimg else None
+            # Perform some checks
+            if mode == "object" and specObjs is None:
+                msgs.warn("No spectral extractions provided for flexure, using slit center instead")
+                mode = "slit"
+            elif mode not in ["object", "slit"]:
+                msgs.warn("mode must be 'object' or 'slit'. Assuming 'slit'.")
```

Comment:
I don't think this should be a warning, but rather it should fault if an unrecognized parameter value is input.